cgnsutilities/cgnsutilities.py

```diff
@@ -1602,34 +1616,46 @@ def coarsen(self):
         conditions and B2B if necessary"""
         # We will coarsen one direction at a time. We do this to check if the block
         # is already 1-cell wide, which can't be coarsened any further
-        if self.dims[0] != 2:
-            self.coords = self.coords[0::2, :, :, :]
-            # Update BCs and B2B
-            for boco in self.bocos:
-                boco.coarsen(0)
-            for b2b in self.B2Bs:
-                b2b.coarsen(0)
 
-        if self.dims[1] != 2:
-            self.coords = self.coords[:, 0::2, :, :]
-            # Update BCs and B2B
-            for boco in self.bocos:
-                boco.coarsen(1)
-            for b2b in self.B2Bs:
-                b2b.coarsen(1)
+        # the new dimensions are half rounded up of the old dimensions
+        new_dims = copy.deepcopy(self.dims)
+        for i in range(3):
+            if self.dims[i] > 2:
+                new_dims[i] = (self.dims[i] + 1) // 2
+
+        new_coords = numpy.zeros((new_dims[0], new_dims[1], new_dims[2], 3))
+
+        # Loop over all directions
+        s = slice(None)
+        fine_slicer = [s] * 3
+        for idx_dim in range(3):
+
+            # can this direction be coarsened?
+            if new_dims[idx_dim] > 2:
+
+                fine_slicer[idx_dim] = slice(None, None, 2)
 
-        if self.dims[2] != 2:
-            self.coords = self.coords[:, :, 0::2, :]
-            # Update BCs and B2B
             for boco in self.bocos:
-                boco.coarsen(2)
+                boco.coarsen(idx_dim)
             for b2b in self.B2Bs:
-                b2b.coarsen(2)
+                b2b.coarsen(idx_dim)
```

Comment:
Shouldn't the `boco` and `b2b` coarsen calls be inside the above if statement?

Reply:
fixed in the new verison